playground/signalr/SignalR.AppHost/Program.cs

```diff
@@ -1,9 +1,41 @@
+using Aspire.Hosting.Azure;
+using Azure.Provisioning;
+using Azure.Provisioning.SignalR;
+
 var builder = DistributedApplication.CreateBuilder(args);
 
-var signalr = builder.AddAzureSignalR("signalr1");
+// Configure Azure SignalR in default mode
+var defaultSignalr = builder.AddAzureSignalR("signalrDefault");
 
 builder.AddProject<Projects.SignalRWeb>("webfrontend")
     .WithExternalHttpEndpoints()
-    .WithReference(signalr);
+    .WithReference(defaultSignalr);
+
+// Configure Azure SignalR in serverless mode
+var serverlessSignalr = builder
+    .AddAzureSignalR("signalrServerless")
+    .ConfigureInfrastructure(infra =>
+    {
+        var resource = infra.GetProvisionableResources().OfType<SignalRService>().First(s => s.BicepIdentifier == infra.AspireResource.GetBicepIdentifier());
+        resource.Features.Add(new SignalRFeature()
+        {
+            Flag = SignalRFeatureFlag.ServiceMode,
+            Value = "Serverless"
+        });
+        // Add required role assignment to invoke REST API for serverless mode
+        var principalTypeParameter = infra.GetProvisionableResources().OfType<ProvisioningParameter>().First(o => o.BicepIdentifier == AzureBicepResource.KnownParameters.PrincipalType);
+        var principalIdTypeParameter = infra.GetProvisionableResources().OfType<ProvisioningParameter>().First(o => o.BicepIdentifier == AzureBicepResource.KnownParameters.PrincipalId);
+        infra.Add(resource.CreateRoleAssignment(SignalRBuiltInRole.SignalRRestApiOwner, principalTypeParameter, principalIdTypeParameter));
+    })
+    .RunAsEmulator()
+    .WithEndpoint("emulator", e =>
+    {
+        e.Port = 64323;
```

Comment:
Why do you need this?

Reply:
My bad, was testing something and forgot to remove it. Let me do it now